docs/topic4.md

```diff
@@ -1,3 +1,170 @@
-# Topic 4 - Frameworks
+# Topic 4 - Javascript - Design patterns
 
-[<- To index](../README.md#title) - [Topic 5 - Project ->](./topic5.md)
\ No newline at end of file
+In this Topic we will focus on learning how JavaScript approaches Object-Oriented programming and
+some patterns to group related functionality together.
+
+If you come from Java, or .NET you will find yourself a little bit lost at the beggining.
+ECMAScript6 provides a layer of syntactic sugar over the previous version (5.1) that is expected to simplify the language.
+But there aren't the classic classes from another Object-Oriented languages
+
+**Documentation**
+
+- [Object-oriented JS](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Object-oriented_JS)
```

Comment:
Bro, the documentation could be moved inside each topic? As it was with the other ones.

Let me know if you are busy to do it

Reply:
Max, esto quedó así porque necesitas un poco de conocimiento de todo para arrancar este tópico.